Clang format

kroosec opened the pull request:
Apply clang-format to the base/ directory. Note that there is no option in clang-format to align successive #define's values (see 1st commit message) so the "//clang-format off/on" comments are needed for those cases only.

To replicate, copy config below to .clang-format, and use the command:
clang-format -style=file -i base/*.{c,h}



---
AlignAfterOpenBracket: Align
AlignConsecutiveAssignments: 'false'
AlignConsecutiveDeclarations: 'false'
AlignEscapedNewlines: Left
AlignOperands: 'true'
AlignTrailingComments: 'true'
AllowAllParametersOfDeclarationOnNextLine: 'false'
AllowShortBlocksOnASingleLine: 'false'
AllowShortCaseLabelsOnASingleLine: 'false'
AllowShortFunctionsOnASingleLine: None
AllowShortIfStatementsOnASingleLine: 'false'
AllowShortLoopsOnASingleLine: 'false'
AlwaysBreakAfterReturnType: All
AlwaysBreakBeforeMultilineStrings: 'false'
BinPackArguments: 'true'
BinPackParameters: 'true'
BreakBeforeBinaryOperators: NonAssignment
BreakBeforeBraces: GNU
BreakBeforeTernaryOperators: 'true'
BreakStringLiterals: 'true'
ColumnLimit: '80'
ContinuationIndentWidth: '2'
DerivePointerAlignment: 'false'
IncludeBlocks: Regroup
IndentCaseLabels: 'false'
IndentWidth: '2'
IndentWrappedFunctionNames: 'false'
KeepEmptyLinesAtTheStartOfBlocks: 'false'
Language: Cpp
MaxEmptyLinesToKeep: '1'
PointerAlignment: Right
ReflowComments: 'true'
SortIncludes: 'true'
SpaceAfterCStyleCast: 'true'
SpaceBeforeAssignmentOperators: 'true'
SpaceBeforeParens: Always
SpaceInEmptyParentheses: 'false'
SpacesInCStyleCastParentheses: 'false'
SpacesInContainerLiterals: 'false'
SpacesInParentheses: 'false'
SpacesInSquareBrackets: 'false'
UseTab: Never

...

bjoernricks commented:
@kroosec you should add the clang format config to the PR too. This allows to use the same config for all users. Btw. adding a template to https://github.com/greenbone/templates for the clang format config would be nice too!

kroosec commented:
Updated the PR:
- Added the .clang-format config file.
- Updated the config file to handle some edge cases.
- Applied the formatting to all directories: base/ util/ gmp/ osp/ tests/
- Added "make format" command.
- Sent a PR for the .clang-format config file to be included in the templates repository. https://github.com/greenbone/templates/pull/9

kroosec commented:
@wiegandm Thanks for the feedback. I have rebased the PR on current Master, and fixed the comment wrapping, also added some spacing for better readability.

i.r.w coding style, we have discussed it during our last team meeting, and everyone agrees with the general output. Most important thing is that we can automate formatting/checking.

jjnicola commented on CMakeLists.txt:
```diff
@@ -44,6 +44,20 @@ endif (NOT BUILD_STATIC)
+
+if (CLANG_FORMAT)
+  message (STATUS "Looking for clang-format... ${CLANG_FORMAT}")
+  add_custom_target(format /usr/bin/clang-format "-i" "./base/*.c" "./gmp/*.c"
```
It should use the variable:
```suggestion
  add_custom_target(format  COMMAND ${CLANG_FORMAT} "-i" "./base/*.c" "./gmp/*.c"
```

kroosec replied:
Fixed both issues. @jjnicola Thanks!

jjnicola commented on CMakeLists.txt:
```diff
@@ -44,6 +44,20 @@ endif (NOT BUILD_STATIC)
+  message (STATUS "Looking for clang-format... ${CLANG_FORMAT}")
+  add_custom_target(format /usr/bin/clang-format "-i" "./base/*.c" "./gmp/*.c"
+                    "./osp/*.c" "./util/*.c" "./base/*.h" "./gmp/*.h" "./osp/*.h"
+                    "./util/*.h")
```
Specify WORKING_DIRECTORY 
```suggestion
                    "./util/*.h" WORKING_DIRECTORY "../")
```
